File: app/src/main/java/com/yegor/tesonet/partyapp/networking/TesonetService.java
package com.yegor.tesonet.partyapp.networking;

import com.yegor.tesonet.partyapp.model.Account;
import com.yegor.tesonet.partyapp.model.Server;
import com.yegor.tesonet.partyapp.model.TokenResponse;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import rx.Observable;

/**
 * Beck-end service
 */
interface TesonetService {

    @POST("tokens")
    Observable<TokenResponse> login(@Body Account account);

    @GET("servers")
    Observable<List<Server>> fetchList();
}

File: app/src/main/java/com/alex/tesonettesttask/logic/network/TesonetService.kt
package com.alex.tesonettesttask.logic.network

import com.alex.tesonettesttask.logic.entities.Server
import com.alex.tesonettesttask.logic.network.request.LoginRequest
import com.alex.tesonettesttask.logic.network.response.AccessTokenResponse
import io.reactivex.Observable
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface TesonetService {

    @POST("v1/tokens")
    fun requestToken(@Body loginRequest: LoginRequest): Observable<Response<AccessTokenResponse>>

    @GET("v1/servers")
    fun getServers(): Observable<Response<List<Server>>>
}